Fix:

error: comparison between pointer and integer ('const char *' and 'char')
--- cegui/src/XMLParserModules/TinyXMLParser/CEGUITinyXMLParser.cpp.orig	2013-01-12 11:07:24.000000000 -0600
+++ cegui/src/XMLParserModules/TinyXMLParser/CEGUITinyXMLParser.cpp	2024-08-11 12:43:46.000000000 -0500
@@ -145,8 +145,9 @@
                 processElement(childNode->ToElement());
                 break;
             case CEGUI_TINYXML_NAMESPACE::TiXmlNode::CEGUI_TINYXML_TEXT:
-                if (childNode->ToText()->Value() != '\0')
-                    d_handler->text((utf8*)childNode->ToText()->Value());
+                const char* value = childNode->ToText()->Value();
+                if (value && *value)
+                    d_handler->text((utf8*)value);
                 break;
 
                 // Silently ignore unhandled node type
